### PR TITLE
improve: use jlink to reduce jdk size in the image

### PR DESCRIPTION
**Pull request by nodece**
### Motivation

Reference: https://adoptium.net/blog/2021/08/using-jlink-in-dockerfiles

Reduce size: 398.23 MB -> 245.23 MB

New image:https://hub.docker.com/layers/nodece/bk/4.17.1-jlink/images/sha256-73c8354a981d7dfefcb1247cf84f6c3e91db2342431749e528ba8468c65be1f5?context=explore

Old image: https://hub.docker.com/layers/apache/bookkeeper/4.17.1/images/sha256-342f384bc21b4d31d5349f741cf831f3c5bb47dfea34557104528ccd9597810f?context=explore

### Changes

- Using jlink to compress the JDK and then copy that to the base image

### CODE REVIEW

**Comment by nodece** [comment]
Could you help to review? @shoothzj @lhotari

**Comment by shoothzj** [comment]
Does this only include the JRE, and exclude JDK tools like jmap, etc.?

**Comment by nodece** [comment]
> Does this only include the JRE, and exclude JDK tools like jmap, etc.?

No. This image includes the JDK tools.

**Review by hezhangjian — APPROVED** [review]

**Comment by nodece** [comment]
Could you help to review? @dlg99

File: docker/Dockerfile [hunk 1]
```diff
@@ -42,7 +56,6 @@ RUN set -x \
     && adduser "${BK_USER}" \
     && apt-get update \
     && apt-get install -y ca-certificates apt-transport-https \
-    && apt-get install -y --no-install-recommends openjdk-17-jdk \
```
Comment (dlg99) [inline docker/Dockerfile:45]: as I understand, this silently/without discussion downgrades jdk from 17 to 11 (included in `eclipse-temurin:11`)

**Review by dlg99 — COMMENTED** [review]
Am I right that this change will downgrade jdk from 17 to 11 in the docker image?

File: docker/Dockerfile (hunk 1 above)
Comment (nodece) [reply docker/Dockerfile:45]: Good catch!

**Comment by nodece** [comment]
> Am I right that this change will downgrade jdk from 17 to 11 in the docker image?

You are right, I should use jdk 17.

**Review by dlg99 — APPROVED** [review]

**Pull request merged**